### PR TITLE
add missing predict docs

### LINKED ISSUE #7144: Lightning Module Test Returns
## 🐛 Bug

I am attempting to test my trained model with my test dataset. After running `Trainer.test()`, an empty dict is returned instead of the dict I am logging. I was previously just returning a dict from `test_epoch_end()` since using `self.log()` never returned anything from `Trainer.test()` for me. I previously had this working on PytorchLightning v1.2.1. After updating to v1.2.8, this no longer works and I am unable to get anything returned from `Trainer.test()`. When I print out the dict in `test_epoch_end()`, everything looks ok but nothing is returned. I have included my test loop below with the return method I used to do commented out. Thanks for the help!

### Expected behavior

Expect to be able to return dicts from `Trainer.test()` for further analysis.

### Environment

 - PyTorch Version (e.g., 1.0): 1.8.1
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): Conda
 - Python version: 3.8.8
 - PytorchLightning Version: 1.2.8

### Additional context

```
def test_step(self, batch, batch_nb):
    data, target = batch
    output = self.forward(data)
    loss = torch.nn.functional.cross_entropy(output, target)
    output = torch.nn.functional.softmax(output, dim=1)
    return {"test_loss": loss, "test_out": output, "test_true": target}

def test_epoch_end(self, outputs):
    loss = torch.mean(torch.tensor([x["test_loss"] for x in outputs]))
    
    test_true = torch.cat([x['test_true'] for x in outputs])
    test_out = torch.cat([x['test_out'] for x in outputs])
    
    out_dict = {"test_loss": loss, "test_out": test_out.cpu().numpy(), "test_true": test_true.cpu().numpy())}
    print(out_dict)
    self.log_dict(out_dict, logger=False)
    # return out_dict
```

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #7144

Any idea why we never had any predict docs? @tchaton 
I'm adding minimal docs here so that at least the API is covered. 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]
lgtm

**Review by s-rog — APPROVED** [review]
Didn't know predict was already implemented!

**Review by akihironitta — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM ! We didn't add it yet as we weren't sure of the API yet. All good now !

**Review by SkafteNicki — APPROVED** [review]
